### PR TITLE
Chore/update ethers

### PR DESCRIPTION
**Pull request by fermentfan**
still waiting for the ethr-did-resolver PR to merge and release, but I built it with a local link and everything runs green.

### CODE REVIEW

**Comment by fermentfan** [comment]
was working on an old state of our fork without noticing... Just merged master into it and resolved all conflicts. We should just make sure we're not including those old commits. Maybe I will squash it later today.

File: tsconfig.json [hunk 1]
```diff
@@ -13,9 +13,6 @@
     "outDir": "lib",
     "strict": true,
     "moduleResolution": "node",
-    "types": [
```
Comment (fermentfan) [inline tsconfig.json:16]: @mirceanis are these needed? My tests in Webstorm wouldn't execute until I removed it and I've never seen that before.

**Comment by fermentfan** [comment]
@mirceanis merge master in

**Comment by fermentfan** [comment]
@mirceanis I switched to the newly released resolver package and it's ready for review :)

File: src/__tests__/config.test.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ describe('configuration', () => {
     const ethrDid = new EthrDID({
       identifier: '0xC662e6c5F91B9FcD22D7FcafC80Cf8b640aed247',
       rpcUrl: 'http://localhost:9585',
-      chainNameOrId: 1337,
+      chainNameOrId: BigInt(1337),
```
Comment (mirceanis) [inline src/__tests__/config.test.ts:8]: By the same reasoning as ethr-did-resolver, the chain ID should be string | number.

It's ok to accept bigint, but the primitives should be the default

File: tsconfig.json (hunk 1 above)
Comment (mirceanis) [reply tsconfig.json:16]: If it causes issues, then it's ok to remove them.

They were added to avoid such issues in the first place

**Review by mirceanis — CHANGES_REQUESTED** [review]
Generally looking great, and I very much appreciate the work.

Some changes are still needed, though, by the same logic as the resolver library

File: src/__tests__/index.test.ts [hunk 3]
```diff
@@ -134,7 +130,7 @@ describe('EthrDID', () => {
         beforeAll(async () => {
           const txHash = await ethrDid.addDelegate(delegate2, {
             delegateType: DelegateTypes.sigAuth,
-            expiresIn: 2,
+            expiresIn: 5,
```
[the remark this answers is not in the document]
Comment (fermentfan) [reply src/__tests__/index.test.ts:133]: Yes that's exactly how I did it. Will double check but they should match 1 to 1 :)


I agree to the smellyness xD

[the remark this answers is not in the document]
Comment (fermentfan) [reply src/__tests__/index.test.ts:133]: checked it out and it matches the sleep in the next test suite

File: src/__tests__/config.test.ts (hunk 2 above)
Comment (fermentfan) [reply src/__tests__/config.test.ts:8]: Looking at the code again I think the field can anyhow not be used with a chainId? It is being fed to the EthrDidResolver and used as the chain name to look into the deployment.ts mapping. The tests are all not checking for the id being resolved to a correct chain name. The test "supports numbered chainId" implicitly falls back to the default being mainnet.

Comment (fermentfan) [reply src/__tests__/config.test.ts]: Changed it to number now again!

**Comment by fermentfan** [comment]
I think I fixed everything now @mirceanis

File: src/index.ts [hunk 4]
```diff
@@ -135,20 +139,16 @@ export class EthrDID {
...
-    return receipt.transactionHash
+    return receipt.hash
   }
 
-  async addDelegate(
-    delegate: string,
-    delegateOptions?: DelegateOptions,
-    txOptions: CallOverrides = {}
-  ): Promise<string> {
+  async addDelegate(delegate: string, delegateOptions?: DelegateOptions, txOptions = {}): Promise<string> {
```
Comment (mirceanis) [inline src/index.ts:151]:
```suggestion
  async addDelegate(delegate: string, delegateOptions?: DelegateOptions, txOptions: Overrides = {}): Promise<string> {
```

**Review by mirceanis — APPROVED** [review]
Looks great, thanks for all the effort!

**Comment by mirceanis** [comment]
@fermentfan looks like linter is complaining about one line.
I also spotted a function where you removed the parameter type for the transaction overrides.

Please fix these 2 issues so we can merge the PR.

I can't fix them on my end since I'm not allowed to update the PR

**Pull request merged**